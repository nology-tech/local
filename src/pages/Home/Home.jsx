import Button from "../../components/Button/Button";
import Layout from "../../components/Layout/Layout";
import Carousel from "../../components/Carousel/Carousel";
import Card from "../../components/Card/Card";
import TextField from "../../components/TextField/TextField";
import "./Home.scss";
import { Link } from "react-router-dom";
import { useEffect, useState, useContext } from "react";
import { getInRadius } from "../../api/placeService";
import { handleAddToUserFavourites } from "../../api/userService";

import UserContext from "../../context/UserContext";
import MapIcon from "../../assets/icons/mapMarkerIcon.svg";

const Home = () => {
  const { user } = useContext(UserContext);
  const [businessArr, setBusinessArr] = useState([]);
  const [searchPostCode, setSearchPostCode] = useState("");

  if (businessArr === null) {
    return <p>loading...</p>;
  }

  const handleSearchInput = (event) => {
    const useSearchInput = event.target.value;
    setSearchPostCode(useSearchInput);
  };

  useEffect(() => {
    getData(searchPostCode);
  }, [searchPostCode]);

  const getData = async (searchPostCode) => {
    if (searchPostCode !== "") {
      const data = await getInRadius(searchPostCode, 5);
      setBusinessArr(data);
    }
  };

  const handleSaveToFavourites = (cardId) => {
    handleAddToUserFavourites(cardId, user);
  };

  const handleSaveToFavourites = (cardId) => {
    handleAddToUserFavourites(cardId, user);
  };

  return (
    <Layout isWithMenu={true}>
      <div className="home__search--bar-container">
        <img
          className="home__map-icon"
          src={MapIcon}
          type="image/svg+xml"
          alt="an icon of a map marker"
        />

        <TextField
          uniqueId="location-search-bar"
          inputType="text"
          modifier="location-search-bar"
          placeholderText="Search by Postcode"
          onChange={handleSearchInput}
        />
      </div>

      <div className="home__map-buttons">
        <Button buttonName="map-navigation-zoom" buttonText="+" />
        <Button buttonName="map-navigation-zoom" buttonText="-" />
        <Link to="./list-view" className="home__link--list-view">
          <Button buttonName="map-navigation" buttonText="List View" />
        </Link>
      </div>

      <h1>Page Heading</h1>
      <h2>Section Heading</h2>
      <h3>Panel Heading</h3>
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Non error
        dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis
        maxime tempora, temporibus beatae voluptas repellat rerum. Dignissimos,
        necessitatibus.
      </p>
      {businessArr?.length > 1 && (
        <Carousel
          componentToDisplay={
            <Card
              cardArray={businessArr}
              page="home"
              primaryButtonOnClick={handleSaveToFavourites}
            />
          }
        />
      )}
    </Layout>
  );
};
export default Home;
