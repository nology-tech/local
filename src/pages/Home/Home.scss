@use "../../styles/variables" as *;

.home {
  height: 100%;
  background-image: url("../../assets/images/Map.svg");
  object-fit: fill;
  display: flex;
  flex-direction: column;
}

.home__map-buttons {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin-right: 1rem;
  justify-content: flex-end;
  height: 20rem;
}

.home__search--bar-container {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  position: relative;
  width: 100%;

  .home__map-icon {
    position: absolute;
    height: 2rem;
    left: 10.43%;
  }
}
