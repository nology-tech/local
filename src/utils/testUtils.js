import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
