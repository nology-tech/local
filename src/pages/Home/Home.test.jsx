import Home from "../../pages/Home/Home";

import { customRender } from "../../utils/testUtils";

it("should render the text on the display", () => {
  const { container } = customRender(<Home />);

  expect(container).toMatchSnapshot();
});
