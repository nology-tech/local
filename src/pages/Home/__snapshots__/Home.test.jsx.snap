// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="header"
    >
      <a
        class="header__link--home"
        href="/"
      >
        <img
          alt="an Icon of an backwards facing arrow"
          src="BackButton.svg"
        />
      </a>
      <div
        class="header__container"
      >
        <div
          class="header__logo-title"
        >
          <img
            alt="an image showing the Local logo"
            src="AppLogo.svg"
          />
          <h1
            class="header__title-local"
          >
            LOCAL
          </h1>
        </div>
      </div>
      <div
        class="menu-icon__container"
        role="button"
      >
        <svg
          alt="Menu-icon"
          class="menu-icon__icon"
          fill="none"
          height="30"
          viewBox="0 0 44 30"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 30H44V25H0V30ZM0 17.5H44V12.5H0V17.5ZM0 0V5H44V0H22H0Z"
            fill="black"
          />
        </svg>
      </div>
    </div>
    <div
      class="layout__menu menu-closed"
    >
      <div
        class="menu"
      >
        <div
          class="menu__header"
        >
          <img
            alt="App Logo"
            class="menu__app-logo"
            src="logo.svg"
          />
          <h1
            class="menu__app-title"
          >
            LOCAL
          </h1>
          <img
            alt="Close Button"
            class="menu__close"
            src="close-btn.svg"
          />
        </div>
        <div
          class="menu__home"
        >
          <a
            class="menu__link"
            href="/home"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Home Logo"
                class="menu__logo"
                src="home.svg"
              />
              <h2
                class="menu__option"
              >
                Home
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__favourites"
        >
          <a
            class="menu__link"
            href="/favourites"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Home Logo"
                class="menu__logo"
                src="favourites.svg"
              />
              <h2
                class="menu__option"
              >
                Favourites
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__info"
        >
          <a
            class="menu__link"
            href="/account"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Account Logo"
                class="menu__logo"
                src="account.svg"
              />
              <h2
                class="menu__option"
              >
                Manage Account
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__about"
        >
          <a
            class="menu__link"
            href="/about"
          >
            <div
              class="menu__page"
            >
              <img
                alt="About Logo"
                class="menu__logo"
                src="about.svg"
              />
              <h2
                class="menu__option"
              >
                About
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="menu__contact"
        >
          <a
            class="menu__link"
            href="/contact"
          >
            <div
              class="menu__page"
            >
              <img
                alt="Contact Logo"
                class="menu__logo"
                src="contact.svg"
              />
              <h2
                class="menu__option"
              >
                Contact
              </h2>
              <img
                alt="Arrow"
                class="menu__arrow"
                src="arrow.svg"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="home__search--bar-container"
    >
      <img
        alt="an icon of a map marker"
        class="home__map-icon"
        src="mapMarkerIcon.svg"
        type="image/svg+xml"
      />
      <div
        class="text-field text-field--location-search-bar"
      >
        <label
          class="text-field__label text-field__label--location-search-bar"
          for="location-search-bar"
        />
        <input
          class="text-field__input text-field__input--location-search-bar"
          id="location-search-bar"
          placeholder="Search by Postcode"
          type="text"
        />
      </div>
    </div>
    <div
      class="home__map-buttons"
    >
      <button
        class="button button--map-navigation-zoom"
      >
        +
      </button>
      <button
        class="button button--map-navigation-zoom"
      >
        -
      </button>
      <a
        class="home__link--list-view"
        href="/list-view"
      >
        <button
          class="button button--map-navigation"
        >
          List View
        </button>
      </a>
    </div>
    <h1>
      Page Heading
    </h1>
    <h2>
      Section Heading
    </h2>
    <h3>
      Panel Heading
    </h3>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Non error dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis maxime tempora, temporibus beatae voluptas repellat rerum. Dignissimos, necessitatibus.
    </p>
  </div>
</div>
`;
